src/test/java/stepdefs/SearchStepDefs.java

```diff
@@ -36,19 +35,40 @@ public void iAmOnTheGoogleUkHomepage() {
     driver.get("https://www.google.co.uk");
   }
   
-  @When("I enter a search term")
-  public void iEnterASearchTerm() {
+  @When("I enter a {string}")
+  public void iEnterA(String string) {
```

Comment:
Here you are not actually utilising the passed in string, so ensure you are entering it in the search field, again I would pass this down to the page object layer method so it looks something like this:
`new SearchPage().enterSearch(string);`